src/tests/test_jsproxy.py

```diff
@@ -1,6 +1,18 @@
 # See also test_typeconversions, and test_python.
 
 
+def test_jsproxy_dir(selenium):
+    selenium.run_js(
+        """
+        window.a = { x : 2 };
+        return pyodide.runPython(`
+            from js import a
+            dir(a)
+        `);
+        """
+    )
```

Comment:
The implicit assertion here was that it doesn't raise. The next test was failing because `dir(a)` always raised an error but it's a bit of an omnibus test and the failure message isn't very specific. (For some reason `PyObject_GenericDir` doesn't exist, so instead you have to say `object.__dir__(a)` which took me several tries to figure out / get right.) 

 Wouldn't hurt to be more specific though.